pkg/frr/utils.go

```diff
@@ -0,0 +1,102 @@
+package frr
+
+func Chunk[E any](values []E, size int) [][]E {
+	if size <= 0 {
+		panic("size must be > 0")
+	}
+
+	var chunks [][]E
+	for remaining := len(values); remaining > 0; remaining = len(values) {
+		if remaining < size {
+			size = remaining
+		}
+
+		chunks = append(chunks, values[:size])
+		values = values[size:]
+	}
+
+	return chunks
+}
+
+func DeleteByIndex[T any](slice []T, index int) []T {
+	sliceLastIndex := len(slice) - 1
+
+	if index != sliceLastIndex {
+		slice[index] = slice[sliceLastIndex]
+	}
+
```

Comment:
```suggestion
slice[index] = slice[sliceLastIndex]
```
I don't think we actually need the check, that's the beauty of it!
If we try to delete the last entry we would just assign the element to itself which is still correct (and probably even faster than the conditional check, not that it matters)